### PR TITLE
ONI UI - Technical Documentation

### PR DESCRIPTION
**Pull request by daortizh**

### CODE REVIEW

File: ipython/dns/UI.md [hunk 1]
```diff
@@ -0,0 +1,592 @@
+# Open Network Insight - Operational Analytics User Interface
+
+Provides tools for interactive visualization, noise filters, white listing, and attack heuristics.
+
+## Table of Content
+
+- [Open Network Insight - Operational Analytics User Interface](#open-network-insight---operational-analytics-user-interface)
+  * [Table of Content](#table-of-content)
```
Comment (NathanSegerlind) [inline ipython/dns/UI.md]: Table of Contents

File: ipython/dns/UI.md [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+
+## Intended Audience
+
+This document is intended for front end developers who want to contribute to our user inteface. To get the most benefit from this guidance, you should have an understanding of the following technologies:
+
+- HTML
+- CSS
+- JavaScript
+	- D3 library
+	- ReactJS+Flux arquitecture
```
Comment (NathanSegerlind) [inline ipython/dns/UI.md]: architecture

File: ipython/dns/UI.md [hunk 3]
```diff
@@ -0,0 +1,592 @@
...
+
+From now on we assume you are familiar with ReactJS+Flux applications.
+
+DNS has the following sections:
+
+1. [Dashboard](#dashboard)
+2. [Threat Investigation](#threat-investigation)
+3. [Storyboard](#storyboard)
+
+#### Dashboard
```
Comment (NathanSegerlind) [inline ipython/dns/UI.md:136]: A sentence or two clarifying what this is, what it does and what the bullets on the list are would be helpful.

File: ipython/dns/UI.md [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+>
+>- SELECT_COMMENT
+>
+>	Broadcasts the comment being selected
+
+##### Components
+
+###### DateInput.react.js
+
+A pipiline level component that allows user to  select a date. After selecting the date this component will start an `UPDATE_DATE` action from [DnsActions](#dnsactions-js).
```
Comment (NathanSegerlind) [inline ipython/dns/UI.md]: pipeline

File: ipython/dns/UI.md [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+
+###### FilterInput.react.js
+
+A component that allows user to filter suspicious data. As soon as the filter changes, this component will start an `UPDATE_FILTER` action from [DnsActions](#dnsactions-js).
+
+###### GridPanelMixin.react.js
+
+A helper component which renders a table. It allows for customization on final components.
+
+While executing render function, this components looks for:
```
Comment (NathanSegerlind) [inline ipython/dns/UI.md]: executing the render function

File: ipython/dns/UI.md [hunk 6]
```diff
@@ -0,0 +1,592 @@
...
+
+Provides tools for interactive visualization, noise filters, white listing, and attack heuristics.
+
+## Table of Content
+
+- [Open Network Insight - Operational Analytics User Interface](#open-network-insight---operational-analytics-user-interface)
+  * [Table of Content](#table-of-content)
+  * [Intended Audience](#intended-audience)
+  * [Getting Started](#getting-started)
+  * [Technical Documentation](#technical-documentation)
```
Comment (NathanSegerlind) [inline ipython/dns/UI.md:11]: I would put the build, run, debug information before the lengthy discussion of the guts of each component

File: ipython/flow/UI.md [hunk 7]
```diff
@@ -0,0 +1,232 @@
+# Open Network Insight - Operational Analytics User Interface
+
+Provides tools for interactive visualization, noise filters, white listing, and attack heuristics.
+
+## Table of Content
```
Comment (NathanSegerlind) [inline ipython/flow/UI.md]: Table of Contents

File: ipython/flow/UI.md [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+
+## Technical Documentation
+
+There are two architectural approaches when it comes to the UI:
+
+1. [Flows: Inline JavaScript](#flows-inline-javascript)
+2. [DNS: ReactJS + Flux](#dns-reactjs--flux)
+
+### Flows: Inline JavaScript
+Back in the days, when ONI was on its early days, flows followed a traditional approach to build its UI, inline JavaScript :( . This will change on next release, but for now, just keep in mind that every JavaScript code is right in between lines of each HTML file.
```
Comment (NathanSegerlind) [inline ipython/flow/UI.md]: No need for the history and roadmap.... just say "in 1.0.1 we do it this way"

(what if we don't have the time to make the changes in the next release? save roadmap info for the roadmap page)

File: ipython/flow/UI.md [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+3. [Edge Investigation](EdgeInvestigation.md)
+4. [Details View](#details-view)
+
+ 
+##### Suspicious Connects
+*ONI/ipython/sconnects\_scores.html*
+
+Data source:
+
+- *ONI/ipython/user/_{DATE}_/flow_scores.csv*
```
Comment (NathanSegerlind) [inline ipython/flow/UI.md:83]: maybe I'm not getting something... but what if the directory ONI ?  is it defined in an environment variable or something?

File: ipython/flow/UI.md [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+
+*ONI/ipython/user/{DATE}/sbdet-{IP}.tsv*
+
+UI Events:
+
+- Window resize event
+- Primary mouse click event
+- D3's Zoom event
+
+#### Injest Summary
```
Comment (NathanSegerlind) [inline ipython/flow/UI.md]: ingest

**Comment by NathanSegerlind** [comment]
looks pretty good... just a few minor comments that should be addressed first

**Comment by EverLoSa** [comment]
LGTM

**Comment by LedaLima** [comment]
LGTM

File: ipython/flow/UI.md (hunk 9 above)
Comment (daortizh) [inline ipython/flow/UI.md:83]: It means the path where ONI was installed. I'll clarify this

**Pull request merged**